src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2340,36 +2327,20 @@ internal override int MetadataArity
                 }
             }
 
-            private readonly PENamedTypeSymbolNonGeneric _underlying;
-            internal override bool IsNativeInt { get; }
-
             internal override NamedTypeSymbol AsNativeInt(bool asNativeInt)
             {
                 Debug.Assert(this.SpecialType == SpecialType.System_IntPtr || this.SpecialType == SpecialType.System_UIntPtr);
 
-                if (IsNativeInt == asNativeInt)
-                {
-                    return this;
-                }
-
-                return asNativeInt ? new PENamedTypeSymbolNonGeneric(this, isNativeInt: true) : _underlying;
+                return asNativeInt ?
+                    (NamedTypeSymbol)new NativeIntegerTypeSymbol(this) :
+                    this;
             }
 
-            // PROTOTYPE: Temporary approach for AsNativeInt().
             internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool> isValueTypeOverrideOpt = null)
```

Comment:
>Equals [](start = 35, length = 6)

If we override Equals, I would expect us to override GetHashCode

Reply:
`GetHashCode()` would simply delegate to the base.

---
In reply to: [370302983](https://github.com/dotnet/roslyn/pull/40994#discussion_r370302983) [](ancestors = 370302983)

Reply:
> GetHashCode() would simply delegate to the base.

I think it benefits readers and code maintainers when Equals and GetHashCode are implemented together and next to each other. Even when implementation is trivial.

---
In reply to: [373735585](https://github.com/dotnet/roslyn/pull/40994#discussion_r373735585) [](ancestors = 373735585,370302983)